### DISCUSSION TITLE
Passing attributes to Asciidoctor

### DISCUSSION THREAD

**Discussion by obfischer** [docToolchain/docToolchain#1093] [Q&A]
Dear all,

is there a way, which allows me to pass attributes to Asciidoctor via the commandline? I would like to extract values like project version and so on and to use them in our documents.

**Comment by rdmueller** [comment]
currently, we don't have a mechanism for doing so.
The best way to workaround this is to create an `.adoc` file with the attributes you need and include it in your docs.
This file could be easily generated.

**Comment by obfischer** [comment]
Hi @rdmueller, 

I found a way to solve my problem. I overhand the needed value via `-Dkey=value` to the docToolchain command and use the value as page suffix. Doing so allows my to generate a unique document title for each version we are publishing to Confluence.

```
doctoolchain . publishToConfluence \
  -DprojectVersion=$(mvn -o -DforceStdout -Dexpression=project.version -q help:evaluate)
```
